cpp/src/io/csv/csv_reader.cu

```diff
@@ -1337,6 +1379,52 @@ struct ConvertFunctor {
   }
 };
 
+/**---------------------------------------------------------------------------*
+ * @brief CUDA kernel iterates over the data until the end of the current field
+ * 
+ * Also iterates over (one or more) delimiter characters after the field.
+ *
+ * @param[in] raw_csv The entire CSV data to read
+ * @param[in] opts A set of parsing options
+ * @param[in] pos Offset to start the seeking from 
+ * @param[in] stop Offset of the end of the row
+ *
+ * @return long position of the last character in the field, including the 
+ *  delimiter(s) folloing the field data
+ *---------------------------------------------------------------------------**/
+__device__ 
+long seekFieldEnd(const char *raw_csv, const ParseOptions opts, long pos, long stop) {
+	bool quotation	= false;
+	while(true){
+		// Use simple logic to ignore control chars between any quote seq
+		// Handles nominal cases including doublequotes within quotes, but
+		// may not output exact failures as PANDAS for malformed fields
+		if(raw_csv[pos] == opts.quotechar){
+			quotation = !quotation;
+		}
+		else if(quotation==false){
+			if(raw_csv[pos] == opts.delimiter){
+				while (opts.multi_delimiter &&
+					raw_csv[pos + 1] == opts.delimiter) {
```

Comment:
Don't you have to check that you're not reading past the end of row?

Reply:
I assumed it's guaranteed that the row ends with the line terminator. Once the character at [pos+1] is equal to the terminator, the loop will exit.

Reply:
We have seen CSV files where the last row does not have a line terminator.  Would that be a problem?

Reply:
Fixed in the latest commit.